### PR TITLE
Relax the nullability check to allow strings

### PR DESCRIPTION
**Pull request by pakrym**
I've made it too strict. We won't be able to ever get the update in.

I filed https://github.com/Azure/autorest.csharp/issues/922 to re-enable but not sure if we ever are going to do it.

### CODE REVIEW

File: src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs [hunk 1]
```diff
@@ -277,7 +277,10 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                                     writer.Append($"continue;");
                                 }
                             }
-                            else if (!property.IsRequired)
+                            else if (!property.IsRequired &&
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs:280]: This is the change. So you don't have to scroll :)

**Review by ShivangiReja — APPROVED** [review]

**Comment by ShivangiReja** [comment]
Can we write a test to this?

File: src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs (hunk 1 above)
Comment (ShivangiReja) [reply src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs:280]: Thank you! :)

**Comment by pakrym** [comment]
I've reenabled one of the existing tests that I made conditional on the configuration previously.

**Pull request merged**